indico/modules/events/registration/fields/simple.py

```diff
@@ -571,30 +572,62 @@ def process_form_data(self, registration, value, old_data=None, billable_items_l
         # always store no-option as empty dict
         if value is None:
             value = {}
+
+        return_value = {}
+
+        if old_data:
+            old_choices_mapping = {x['id']: x for x in old_data.field_data.versioned_data['choices']}
+            new_choices_mapping = {x['id']: x for x in self.form_item.versioned_data['choices']}
+            revoked_selected_choice_ids = []
```

Comment:
And you can simply create it like this:

    revoked_selected_choice_ids = value.viewkeys() - new_choices_mapping.viewkeys()

*keys from value that are not in new_choices_mapping*